generator/osm2meta.cpp

```diff
@@ -1 +1,39 @@
 #include "generator/osm2meta.hpp"
+
+#include "base/logging.hpp"
+
+string MetadataTagProcessor::ValidateAndFormat_wikipedia(string const & v) const
+{
+  // Normalize by converting full URL to "lang:title" if necessary
+  // (some OSMers do not follow standard for wikipedia tag).
+  static string const base = ".wikipedia.org/wiki/";
+  auto const baseIndex = v.find(base);
+  if (baseIndex != string::npos)
+  {
+    auto const baseSize = base.size();
+    if (v.size() > baseIndex + baseSize)
+    {
+      auto const slashIndex = v.rfind('/', baseIndex);
+      if (slashIndex != string::npos && slashIndex + 1 != baseIndex)
+        return v.substr(slashIndex + 1, baseIndex - slashIndex - 1)
+            + ":" + v.substr(baseIndex + baseSize);
+    }
+    LOG(LWARNING, ("Invalid Wikipedia tag value:", v));
```

Comment:
Already tried it. Too many duplicated LOG(LWARNING... lines are harder to read than a bit wider indent.